docs - dashboard subscriptions - closes #14075 (#14209)
metabase/metabase: docs/users-guide/dashboard-subscriptions.md

## Dashboard subscriptions

> Dashboard subscriptions are a great way to keep you and your team up to date on the data that matters most. They allow you to send all of the questions on a dashboard via email or Slack. If your Metabase has email or Slack set up, all you need to do is create a dashboard, add subscribers to it, and tell Metabase how often you'd like the send out an update. You can set up as many subscriptions to a dashboard as you like, and if you make any changes to the dashboard, Metabase will update the subscription the next time they're delivered.

Reviewer: (or "the subscription the next time **it's** delivered")

### Enabling dashboard subscriptions

To enable dashboard subscriptions, your administrators will need to have set up email or Slack for your Metabase. See [Setting up email](https://www.metabase.com/docs/latest/administration-guide/02-setting-up-email.html) or [Setting up Slack](https://www.metabase.com/docs/latest/administration-guide/09-setting-up-slack.html).

### Setting up a dashboard subscription

To set up a subscription to a dashboard, click on the **sharing** icon (the one-way arrow) and select **Dashboard subscriptions**.

![Select dashboard subscriptions](images/dashboard-subscriptions/select-dashboard-subscription.png)

Metabase will slide out a sidebar on the right, with an option set up a subscription via email or Slack:

![Set up a dashboard subscription with email or slack](images/dashboard-subscriptions/email-or-slack.png)

Let's say we want to email a dashboard. We'll click on the **Email it** option in the sidebar, and Metabase will give us some options:

![Dashboard subscription email options](images/dashboard-subscriptions/email-options.png)

### Email subscription options

> For emails, we can:

Reviewer 2: I'd advocate for listing the options in the order they appear. I think the send-now button's bullet and the attach results bullet are in the wrong positions.

- **Add subscribers**. We can add subscribers to the dashboard by adding their email addresses.
- **Determine frequency and timing**. Determine how often Metabase sends the dashboard (daily, weekly, or monthly) and what time of day Metabase sends the dashboard.

> - **Attach results**. Whether Metabase should also attach results to the email (which will include up 2000 rows of data). Note that Metabase will automatically attach results as a table for questions that do not translate

Reviewer 2: Looks like this is an unfinished paragraph.

Reviewer 2: You can also choose between csv or xlsx for your attachments' file format.

- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the dashboard at all.

> - **Immediate gratification:** Or send an email right now, which is great for sending a test email.

Reviewer 2: I appreciate the humor but don't know if this is clear, and leaves me with questions about the actual functionality of the send-now button, e.g., does this test email get sent to me, to all the recipients, or…?

### Email example

![Example dashboard subscription email](images/dashboard-subscriptions/example-email.png)

> You'll notice in the email that Metabase excludes any text cards on the dashboard, and that the charts look different - Metabase reformats the charts to make them more legible in email. Additionally, tables that exceed either 10 columns or 20 rows get the rest of their results, up to 2000 rows, included as an attachment. As a bonus, Metabase will toss in a random data quotation, like this one from the mathematician Andrejs Dunkels: "It's easy to lie with statistics. It's hard to tell the truth without statistics."

Reviewer 2: Dunno that we need to call out the quotation bit. Additionally, once we get the click/linking behavior dialed in, that's something we might want to call out — "Clicking on a question in the email will take you to the dashboard."

```suggestion
You'll notice in the email that Metabase excludes any text cards on the dashboard, and that the charts look different - Metabase reformats the charts to make them more legible in email. Additionally, tables that exceed either 10 columns or 20 rows get the rest of their results, up to 2000 rows, included as an attachment.
```

> ### Slack subscription options

Reviewer 2: Slack subscriptions currently also have an additional schedule option to send hourly I believe.

> For Slack subscriptions, you can set up a subscription for a single person using their Slack username, or for a channel (like #general).

Reviewer 2: I think you've got the importance reversed here — seems like the more common usecase is sending a slack subscription to a channel. E.g., "You can also send a subscription to yourself or another user in Slack."

![Select dashboard subscriptions](images/dashboard-subscriptions/select-dashboard-subscription.png)

Metabase will slide out a sidebar on the right, with an option to set up a subscription via email or Slack:

![Set up a dashboard subscription with email or slack](images/dashboard-subscriptions/email-or-slack.png)

Let's say we want to email a dashboard. We'll click on the **Email it** option in the sidebar, and Metabase will give us some options:

![Dashboard subscription email options](images/dashboard-subscriptions/email-options.png)

### Email subscription options

For emails, we can:

- **Add subscribers**. Add email addresses to register subscribers.
- **Determine frequency and timing**. Tell Metabase how often it should send the dashboard (daily, weekly, or monthly), and what time of day to send the dashboard.
- **Send email now** sends an email to all subscribers.
- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the email.
- **Attach results**. Tell Metabase if it should also attach results to the email (which will include up to 2000 rows of data). You can choose between CSV and XLSX file formats.

> If you've added filters to your dashboard and set default values for those filters, the subscriptions will run the dashboard with those default values applied, filtering results for all questions with fields connected to those filters. To learn more, check out [dashboard filters](https://www.metabase.com/docs/latest/users-guide/08-dashboard-filters.html).

Reviewer 2: ```suggestion
If you've added filters to your dashboard and set default values for those filters, those default values will be applied to your subscriptions, filtering the results of all questions that are connected to those filters when the subscriptions are sent. To learn more, check out [dashboard filters](https://www.metabase.com/docs/latest/users-guide/08-dashboard-filters.html).
```
